content/en/docs/setup/additional-setup/sidecar-injection/index.md

```diff
@@ -153,202 +153,94 @@ sleep-776b7bcdcd-bhn9m   2/2       Terminating   0          2m
-      DNSConfig           *corev1.PodDNSConfig          `yaml:"dnsConfig"`
-      ImagePullSecrets    []corev1.LocalObjectReference `yaml:"imagePullSecrets"`
-}
-{{< /text >}}
-
-The template is applied to the following data structure at runtime.
-
-{{< text go >}}
-type SidecarTemplateData struct {
-    DeploymentMeta *metav1.ObjectMeta
-    ObjectMeta     *metav1.ObjectMeta
-    Spec           *corev1.PodSpec
-    ProxyConfig    *meshconfig.ProxyConfig  // Defined by https://istio.io/docs/reference/config/service-mesh.html#proxyconfig
-    MeshConfig     *meshconfig.MeshConfig   // Defined by https://istio.io/docs/reference/config/service-mesh.html#meshconfig
-}
-{{< /text >}}
-
-`ObjectMeta` and `Spec` are from the pod. `ProxyConfig` and `MeshConfig`
-are from the `istio` ConfigMap in the `istio-system` namespace. Templates can conditionally
-define injected containers and volumes with this data.
-
-For example, the following template
-
-{{< text plain >}}
-containers:
-- name: istio-proxy
-  image: istio.io/proxy:0.5.0
-  args:
-  - proxy
-  - sidecar
-  - --configPath
-  - {{ .ProxyConfig.ConfigPath }}
-  - --binaryPath
-  - {{ .ProxyConfig.BinaryPath }}
-  - --serviceCluster
-  {{ if ne "" (index .ObjectMeta.Labels "app") -}}
-  - {{ index .ObjectMeta.Labels "app" }}
-  {{ else -}}
-  - "istio-proxy"
-  {{ end -}}
-{{< /text >}}
-
-expands to
-
-{{< text yaml >}}
-containers:
-- name: istio-proxy
-  image: istio.io/proxy:0.5.0
-  args:
-  - proxy
-  - sidecar
-  - --configPath
-  - /etc/istio/proxy
-  - --binaryPath
-  - /usr/local/bin/envoy
-  - --serviceCluster
-  - sleep
-{{< /text >}}
-
-when applied over a pod defined by the pod template spec in [`samples/sleep/sleep.yaml`]({{< github_tree >}}/samples/sleep/sleep.yaml)
-
-#### More control: adding exceptions
+  containers:
+  - name: hello
+    image: alpine
+  - name: istio-proxy
+    image: auto
+    resources:
+      requests:
+        cpu: "100m"
+    volumeMounts:
+    - mountPath: /etc/certs
+      name: certs
+    lifecycle:
+      preStop:
+        exec:
+          command: ["sleep", "10"]
+  volumes:
+  - name: certs
+    secret:
+      secretName: istio-certs
+{{< /text >}}
+
+In general, any field in a pod can be set. However, care must be taken for certain fields:
+
+* Kubernetes requires the `image` field to be set before the injection has run. While you can set a specific image to override the default one,
+  it is recommended to set the `image` to `auto` which will cause the sidecar injector to automatically select the image to use.
+* Some fields in `Pod` are dependent on related settings. For example, CPU request must be less than CPU limit. If both fields are not configured together, the pod may fail to start.
+
+Additionally, certain fields are configurable by [annotations](/docs/reference/config/annotations/) on the pod, although it is recommend to use the above approach to customizing settings.
```

Comment:
I think this is confusing.  Would not we recommend users to use annotations first, and if that doesn't provide the customization they need, they can look at the customization via pod yaml as described here.

Reply:
I don't see any reason to use the annotations now that we have the new way. WDYT?

Reply:
I am not sure how to do proxy config annotation using the new way?  for example, if i want to use holdApplicationUntilProxyStarts.

Reply:
that one is an annotation :stuck_out_tongue: . I will include that in a new PR to clarify